src/App.js

```diff
@@ -10,10 +12,34 @@ const tabs = [
 ];
 /* eslint-enable */
 
-const App = () => (
-  <div className="App">
-    <h1>Selected tab is&nbsp;Tab 1</h1>
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedTab: tabs[0],
+  }
+
+  onTabSelect = (tab) => {
+    this.setState({
+      selectedTab: tab,
+    });
+  }
```

Comment:
If user clicks the tab that is already selected, `setState` should not be called.